src/main/java/com/example/demo/repository/UserRepository.java

```diff
@@ -0,0 +1,11 @@
+package com.example.demo.repository;
+
+import com.example.demo.entity.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Optional;
+import java.util.UUID;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+  Optional <User> findByusername(String username);
```

Comment:
this should be named as findByUsername according to the convention of spring data jpa